cmd/ref_link_ensurer.go

```diff
@@ -0,0 +1,42 @@
+package cmd
+
+import (
+	"github.com/qri-io/qri/dsref"
+	"github.com/qri-io/qri/lib"
+)
+
+// EnsureFSIAgrees should be passed to GetCurrentRefSelect in order to ensure that any references
+// used by a command have agreement between what their .qri-ref linkfile thinks and what the
+// qri repository thinks. If there's a disagreement, the linkfile wins and the repository will
+// be updated to match.
+// This is useful if a user has a working directory, and then manually deletes the .qri-ref (which
+// will unlink the dataset), or renames / moves the directory and then runs a command in that
+// directory (which will update the repository with the new working directory's path).
+func EnsureFSIAgrees(f *lib.FSIMethods) *FSIRefLinkEnsurer {
```

Comment:
Kinda a funny API, but it gets the job done! I'd say if we have to add another option fo `GetCurrentRefSelect`, we should consider re-structuring, I'd rather not force an API re-think right now

Reply:
This behavior has always existed, since EnsureRef was a thing, however it wasn't explicit. Callers would pass an fsiMethods pointer in, but it was unclear what that meant. I've seen at least one PR author confused by this fact. This type constructor clarifies the callsites of GetRefSelect a bit.